common/beerocks/bwl/CMakeLists.txt

```diff
@@ -91,19 +100,29 @@ elseif(BWL_TYPE STREQUAL "NL80211")
     )
 
     # Hostapd/NL80211 include directories
-    include_directories(
+    list(APPEND BWL_INC_DIRS
         ${HOSTAPD_DIR}/src/utils
         ${HOSTAPD_DIR}/src/common
         ${HOSTAPD_DIR}/src/drivers
-        ${PLATFORM_INCLUDE_DIR}/libnl-tiny
     )
+    # OpenWrt support only libnl-tiny
+    if (TARGET_PLATFORM STREQUAL "openwrt")
+        list(APPEND BWL_INC_DIRS ${PLATFORM_INCLUDE_DIR}/libnl-tiny)
+    endif()
+    include_directories(${BWL_INC_DIRS})
 
     # Platform libraries
     link_directories(
         ${PLATFORM_STAGING_DIR}/usr/lib
     )
 
-    list(APPEND BWL_LIBS nl-tiny)
+    if (TARGET_PLATFORM STREQUAL "openwrt")
+        list(APPEND BWL_LIBS nl-tiny wpa_client)
+    else()
+        find_package(nl-3 REQUIRED)
+        find_package(nl-genl-3 REQUIRED)
+        list(APPEND BWL_LIBS nl-3 nl-genl-3 wpa_client)
```

Comment:
wpa_client should not be added here, but in the NEEDS_HOSTAPD_PATH condition that I mentioned above. You could first check if it exists, like this:
```
find_library(WPA_CLIENT_LIB libwpa_client.so)
if (${WPA_CLIENT_LIB} STREQUAL WPA_CLIENT_LIB-NOTFOUND)
    ... do the hostapd_dir stuff
else()
    list(APPEND BWL_LIBS wpa_client)
endif()
```